lexer_test.go

```diff
@@ -27,28 +25,26 @@ import (
 )
 
 const (
-	unusedType TokenType = iota
-	wordType
+	wordType TokenType = iota + 1
 )
 
 type lexWordState struct{}
 
 func (w *lexWordState) Run(_ context.Context, l *Lexer) (LexState, error) {
-	rn, err := l.Peek(1)
-	if errors.Is(err, io.EOF) || (err == nil && unicode.IsSpace(rn[0])) {
+	rn := l.Peek()
+	if unicode.IsSpace(rn) {
 		// NOTE: This can emit empty words.
 		l.Emit(wordType)
 		// Discard the space
-		if _, dErr := l.Discard(len(rn)); dErr != nil {
-			return nil, dErr
+		if !l.Discard() {
+			return nil, nil
 		}
 	}
-	if err != nil {
-		return nil, err
-	}
 
-	if _, aErr := l.Advance(len(rn)); aErr != nil {
-		return nil, aErr
+	if !l.Advance() {
+		// Emit if we hit the end of input.
+		l.Emit(wordType)
+		return nil, nil
 	}
```

Comment:
This is a bit shorter and maybe clearer about EOF handling.
```go
rn := l.Peek()
	if unicode.IsSpace(rn) || rn == EOF {
		// NOTE: This can emit empty words.
		l.Emit(wordType)
		// Discard the space
		if !l.Discard() {
			return nil, nil
		}
	}

	l.Advance()
```